Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs

```diff
@@ -4152,53 +4352,125 @@ public void HideKeyboard()
                 else if (Driver is IOSDriver)
                 {
                     ((IOSDriver)Driver).HideKeyboard();
+                    // ((IOSDriver)Driver).FindElement(By.Name("Cancel")).Click();
+                    //((IOSDriver)Driver).FindElement(By.Name("Done")).Click();
+                    //((IOSDriver)Driver).FindElement(By.Name("Next:")).Click();
+                    //((IOSDriver)Driver).FindElement(By.Name("id.ui.browse.back.button")).Click();
+                    //var outsideElement = Driver.FindElement(By.XPath("//XCUIElementTypeButton"));
+                    //outsideElement.Click();
+                    //var size = Driver.Manage().Window.Size;
+                    //int centerX = size.Width / 2;
+                    //int centerY = size.Height / 2;
+
+                    //var touchAction = new OpenQA.Selenium.Interactions.Actions(Driver);
+                    //touchAction.MoveToLocation(centerX, centerY).Click().Perform();
+                }
+                else
+                {
+                    throw new NotSupportedException($"Driver type '{Driver.GetType().Name}' is not supported for keyboard operations");
                 }
             }
             catch (Exception ex)
             {
-                throw new($"Failed to hide the keyboard: {ex.Message}");
+                throw new InvalidOperationException($"Failed to hide the keyboard: {ex.Message}", ex);
             }
         }
 
         public void PushFileToDevice(string LocalFilePath, string DeviceTargerFolder)
         {
-            byte[] fileContent = System.IO.File.ReadAllBytes(LocalFilePath);
-            string fileName = Path.GetFileName(LocalFilePath);
-            if (Driver is IOSDriver)
+            if (string.IsNullOrEmpty(LocalFilePath))
+                throw new ArgumentException("Local file path cannot be null or empty", nameof(LocalFilePath));
+
+            if (string.IsNullOrEmpty(DeviceTargerFolder))
+                throw new ArgumentException("Device target folder cannot be null or empty", nameof(DeviceTargerFolder));
+
+            //if (!File.Exists(LocalFilePath))
+            //    throw new ArgumentException($"Local file not found: {LocalFilePath}", nameof(LocalFilePath));
+            bool isDirectory = Directory.Exists(LocalFilePath);
+            bool isFile = File.Exists(LocalFilePath);
+            if (!isDirectory && !isFile)
+                throw new ArgumentException($"Path not found: {LocalFilePath}", nameof(LocalFilePath));
+
+            if (isDirectory)
             {
-                ((IOSDriver)Driver).PushFile($"{DeviceTargerFolder}/{fileName}", fileContent);
+                // Handle directory
+                PushFolder(LocalFilePath, DeviceTargerFolder);
```

Comment:
instead having method for File and for Folder/multi Files so combine into one method which will get "FilesPath", if it folder so pull all files in it to a list (Directory.GetFiles()) and if single file so add it also to the list, then next code will loop pn the List and push the files one by one from the list